modelon/impact/client/client.py

```diff
@@ -71,7 +72,16 @@ def _authenticate_against_api(self, interactive):
         else:
             login_data = {"secretKey": self.api_key}
 
-        self._sal.api_login(login_data)
+        try:
+            self._sal.api_login(login_data)
+        except modelon.impact.client.sal.exceptions.HTTPError as error:
+            if interactive:
+                self._credentials.delete_key()
+            raise modelon.impact.client.exceptions.InvalidAPIKeyError(
```

Comment:
We should probably skip raising this specific error from the generic http error. We should do something with error codes for more precise errors in the future, but for now we should not try to make to many assumptions that risk introducing bugs/confusion.